### DISCUSSION TITLE
Problem: One vercel's postgres, and need a way to use it in multiple projects

### DISCUSSION THREAD

**Discussion by iyxan23** [prisma/prisma#22517] [Q&A]
### Question

Hey all!

I'm just going to go to the point, I have a single free vercel plan that only allows to have one postgres database. I'm thinking of using that one database for multiple projects at once with each its own different prisma schemas.

I've renamed all each and every of the tables on each schemas to have a prefix. But as I tried to deploy to vercel (which has `prisma db push` as its postinstall script), prisma asked me to drop these tables.

Is there a way to prevent prisma from deleting these tables, and just let them be?

![image](https://github.com/prisma/prisma/assets/31884435/61728983-3232-4e41-b714-6574afef90e6)

(there are no conflicting table names)

Would really appreciate quick answers, I'm quite in a hurry right now! Thank you!

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: Fedora 38
- Database: PostgreSQL
- Node.js version: v20.8.1
- Prisma version dump:
  ```
  prisma                  : 5.4.2
  @prisma/client          : 5.4.2
  Current platform        : rhel-openssl-1.1.x
  Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/libquery_engine-rhel-openssl-1.1.x.so.node)
  Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/schema-engine-rhel-openssl-1.1.x)
  Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
  Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
  Studio                  : 0.494.0
  ```

**Comment by janpio** [comment]
No. You will probably have better success using one schema per project.

**Comment by keeandev** [comment]
> ### Question
> Hey all!
> 
> I'm just going to go to the point, I have a single free vercel plan that only allows to have one postgres database. I'm thinking of using that one database for multiple projects at once with each its own different prisma schemas.
> 
> I've renamed all each and every of the tables on each schemas to have a prefix. But as I tried to deploy to vercel (which has `prisma db push` as its postinstall script), prisma asked me to drop these tables.
> 
> Is there a way to prevent prisma from deleting these tables, and just let them be?
> 
> ![image](https://private-user-images.githubusercontent.com/31884435/292975219-61728983-3232-4e41-b714-6574afef90e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MjY1OTEsIm5iZiI6MTcxMzkyNjI5MSwicGF0aCI6Ii8zMTg4NDQzNS8yOTI5NzUyMTktNjE3Mjg5ODMtMzIzMi00ZTQxLWI3MTQtNjU3NGFmZWY5MGU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDAyMzgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjZWExZWJmY2Y2ZGQzYWU3MDI3YmIzZWNlYTRjOGMxY2FiNTE5OTA3M2E5NDhhZmQ1Njk4ZWM1NmVjNzIwMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wcW1-0ltNpvfSIZ-NuSlEHH1w-VWQ06GGzMHagyK4V8)
> 
> (there are no conflicting table names)
> 
> Would really appreciate quick answers, I'm quite in a hurry right now! Thank you!
> 
> ### How to reproduce (optional)
> ### Expected behavior (optional)
> _No response_
> 
> ### Information about Prisma Schema, Client Queries and Environment (optional)
> * OS: Fedora 38
> * Database: PostgreSQL
> * Node.js version: v20.8.1
> * Prisma version dump:
>   ```
>   prisma                  : 5.4.2
>   @prisma/client          : 5.4.2
>   Current platform        : rhel-openssl-1.1.x
>   Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/libquery_engine-rhel-openssl-1.1.x.so.node)
>   Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/schema-engine-rhel-openssl-1.1.x)
>   Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
>   Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
>   Studio                  : 0.494.0
>   ```

If you're looking to solve this issue a different way, try this out: https://orm.drizzle.team/docs/goodies#multi-project-schema

**Reply by iyxan23** [reply]
yes! I switched to drizzle and it was a breeze. In case anyone else is attempting to do the same thing, use drizzle instead.

**Reply by keeandev** [reply]
Totally!

**Reply by KadaGuetouache** [reply]
I haven't tried it yet but also prisma allows for multi-project-schema [prisma-muliple-schema](https://www.prisma.io/docs/orm/prisma-schema/data-model/multi-schema)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
